pkg/controller/disruption/disruption_test.go

```diff
@@ -390,30 +415,41 @@ func TestMultipleControllers(t *testing.T) {
 	pdb, pdbName := newPodDisruptionBudget(t, intstr.FromString("1%"))
 	add(t, dc.pdbStore, pdb)
 
+	pods := []*v1.Pod{}
 	for i := 0; i < podCount; i++ {
 		pod, _ := newPod(t, fmt.Sprintf("pod %d", i))
+		pods = append(pods, pod)
 		add(t, dc.podStore, pod)
 	}
+
 	dc.sync(pdbName)
 
 	// No controllers yet => no disruption allowed
 	ps.VerifyDisruptionAllowed(t, pdbName, 0)
 
 	rc, _ := newReplicationController(t, 1)
 	rc.Name = "rc 1"
+	for i := 0; i < podCount; i++ {
+		updatePodOwnerToRc(t, pods[i], rc)
```

Comment:
This PR LGTM other than the potential flakiness here. If you want, we can just try to measure whether this is going to be flaky and submit it as is if it passes.

One way to do that is something like `go test -count 1000` or `go test -race -count 1000`. You might want to start with a smaller count and work up to what your machine can handle. It seems that Go sometimes runs the tests concurrently and I've locked up my workstation before by putting a count that was too high. :)